src/Surfnet/Stepup/Identity/Value/SecondFactorIdentifier.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Identity\Value;
+
+interface SecondFactorIdentifier
```

Comment:
I could remove `::unknown()`, as it is never called dynamically.

Are you suggesting I replace calls to `equals()`, `__toString()` with `getValue()->...()`, `(string) $sfi->getValue()`? I do like being able to check different SecondFactorIdentifiers for equality. Let's discuss IRL.

Also, what are your thoughts about moving the SecondFactorType into the SecondFactorIdentifier? Their types already imply their... types.